src/vs/workbench/common/editor/untitledEditorInput.ts

```diff
@@ -87,8 +86,13 @@ export class UntitledEditorInput extends AbstractUntitledEditorInput {
 			return this.cachedModel.isDirty();
 		}
 
-		// untitled files with an associated path or restore resource are always dirty
-		return this.hasAssociatedFilePath || !!this.restoreResource;
+		// A disposed input is never dirty, even if it was restored from backup
+		if (this.isDisposed()) {
+			return false;
+		}
+
+		// untitled files with an associated path or associated resource
+		return this.hasAssociatedFilePath || !!this.resource;
```

Comment:
@Tyriar isn't that wrong? every untitled editor input will have a resource but will not be dirty when opened (e.g. open a new untitled file without content, it should not show dirty).